components/contact.tsx

```diff
@@ -0,0 +1,297 @@
+"use client";
+import Link from "next/link";
+import { useState } from "react";
+import { signup, signinWithOAuth, contactSubmit } from "@/app/(auth)/actions";
+import { Provider } from "@supabase/supabase-js";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Select } from "@/components/ui/select";
+import {
+  Form,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormControl,
+  FormMessage,
+} from "@/components/ui/form";
+import { contactSchema, ContactFormData } from "@/utils/form-schema";
+import { toast } from "sonner";
+import { useRouter } from "next/navigation";
+
+const organisationSizes = [
+  { value: "1-10", label: "1-10 employees" },
+  { value: "11-50", label: "11-50 employees" },
+  { value: "51-200", label: "51-200 employees" },
+  { value: "201-500", label: "201-500 employees" },
+  { value: "501-1000", label: "501-1000 employees" },
+  { value: "1001+", label: "1001+ employees" },
+];
+
+export default function ContactUs() {
+  const [isSubmitting, setIsSubmitting] = useState(false);
```

Comment:
I recommend using the `isSubmitting` status provided in `formState` by `react-hook-form` to avoid manually setting the update status.